actors/builtin/miner/miner_actor.go

```diff
@@ -342,6 +318,16 @@ func (a Actor) PreCommitSector(rt Runtime, params *SectorPreCommitInfo) *adt.Emp
 	if params.Expiration <= rt.CurrEpoch() {
 		rt.Abortf(exitcode.ErrIllegalArgument, "sector expiration %v must be after now (%v)", params.Expiration, rt.CurrEpoch())
 	}
+	if params.SealRandEpoch >= rt.CurrEpoch() {
```

Comment:
@magik6k is going to want to work this into the fsm

Reply:
As I understand it, `SealRandEpoch` is the epoch from which the challenge in seal phase 1 is sampled, producing the CommR which is also presented in this params. So it would make no sense for the challenge to be sampled after the precommit (which is the current epoch).  

The randomness isn't actually _checked_ until ProveCommit, though. At ProveCommit we check that it is no older than some window, but if the miner could arbitraily forward-date this that check wouldn't be doing anything.

But if I have that wrong, would love to be explained otherwise (and you can just drop this precondition).

Reply:
This should be fine